crowbar_framework/app/controllers/backups_controller.rb

```diff
@@ -0,0 +1,171 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class BackupsController < ApplicationController
+  skip_before_filter :enforce_installer
+  before_action :set_backup, only: [:destroy, :restore, :download]
+
+  #
+  # Backups
+  #
+  # Provides the restful api call for
+  # /utils/backup 	GET 	Returns a json list of available backups
+  def index
+    @backups = Backup.all
+
+    respond_to do |format|
+      format.html
+      format.json { render json: @backups }
+    end
+  end
+
+  #
+  # Backups
+  #
+  # Provides the restful api call for
+  # /utils/backup   POST   Trigger a backup
+  def create
+    @backup = Backup.new(backup_params)
+
+    respond_to do |format|
+      if @backup.save
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        msg = I18n.t("backups.index.invalid_filename")
+        format.json { render json: { error: msg }, status: :bad_request }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Restore
+  #
+  # Provides the restful api call for
+  # /utils/backup/restore   POST   Trigger a restore
+  def restore
+    respond_to do |format|
+      if @backup.restore
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        format.json do
+          render json: { error: @backup.errors.full_messages.first }, status: :unprocessable_entity
+        end
+        format.html do
+          flash[:alert] = @backup.errors
```

Comment:
Wrong... This should be some generic i18n message like on create. `errors` is a hash of error messages, so this should never work.